app/colorCoder.js

```diff
@@ -1,99 +1,40 @@
-{
-	let minorSize = MajorColorNames.length;
-	let majorSize = MinorColorNames.length;
-	
-	if (pairNumber < 1 || pairNumber > minorSize * majorSize)
- 	{
- 		throw `Argument PairNumber:${pairNumber} is outside the allowed range` 
-	}
-	let zeroBasedPairNumber = pairNumber - 1;
-    let majorIndex = parseInt (zeroBasedPairNumber / minorSize);
-    let minorIndex = parseInt(zeroBasedPairNumber % minorSize);
-	let  pair = new ColorPair();
-	pair.majorColor = MajorColorNames[majorIndex];
-	pair.minorColor = MinorColorNames[minorIndex];
-	return pair;
-}
-
-function getPairNumberFromColor(pair)
-        {
-        let majorIndex = -1;
-        for (let i = 0; i < MajorColorNames.length; i++)
-            {
-                if (MajorColorNames[i] == pair.majorColor)
-                {
-                    majorIndex = i;
-                    break;
-                }
-            }
-
-        let minorIndex = -1;
-            for (let i = 0; i < MinorColorNames.length; i++)
-            {
-                if (MinorColorNames[i] == pair.minorColor)
-                {
-                    minorIndex = i;
-                    break;
-                }
-            }
-    
-        if (majorIndex == -1 || minorIndex == -1)
-            {
-                throw `Unknown Colors:${pair.toString()}`;
-            }
-
-        return (majorIndex * MinorColorNames.length) + (minorIndex + 1);
-	 }
-	 
-	 function test(){
-			pairNumber = 4;
-            let testPair1 = getColorFromPairNumber(pairNumber);
-            console.log(`[In]Pair Number: ${pairNumber},[Out] Colors:${testPair1}`);
-			console.assert(testPair1.majorColor == "WHITE");
-			console.assert(testPair1.minorColor == "BROWN");
-
-			pairNumber = 5;
-            testPair1 = getColorFromPairNumber(pairNumber);
-            console.log(`[In]Pair Number: ${pairNumber},[Out] Colors:${testPair1}`);
-			console.assert(testPair1.majorColor == "WHITE");
-			console.assert(testPair1.minorColor == "SLATEGRAY");
-			 
-			pairNumber = 23;
-            testPair1 = getColorFromPairNumber(pairNumber);
-            console.log(`[In]Pair Number: ${pairNumber},[Out] Colors:${testPair1}`);
-			console.assert(testPair1.majorColor == "RED");
-			console.assert(testPair1.minorColor == "GREEN");
-			 
-            let testPair2 = new ColorPair();
-			testPair2.majorColor="YELLOW";
-			testPair2. minorColor ="GREEN";
-            pairNumber =getPairNumberFromColor(testPair2);
-            console.log(`[In]Colors: ${testPair2}, [Out] PairNumber: ${pairNumber}`);
-			console.assert(pairNumber==18);
-
-			testPair2 = new ColorPair();
-			testPair2.majorColor="RED";
-			testPair2. minorColor ="BLUE";
-            pairNumber =getPairNumberFromColor(testPair2);
-            console.log(`[In]Colors: ${testPair2}, [Out] PairNumber: ${pairNumber}`);
-			console.assert(pairNumber==6);
+let ColorPair = require('./color_pair');
+let conversion = require('./conversion');
+const print_manual = require('./manual');
+
+function test(){
```

Comment:
The filename `colorCoder.js` doesn't give any hint that it's purpose is to test the functionality.

Reply:
I have modified the filename to colorCoderTest.js. Ref: Commit ID e832e967f57d3fd3198fcdd3d1afabde7ffd258e